### ISSUE TITLE
[css-image-animation] Conveying role, states, properties to platform AAPIs

### ISSUE THREAD

**Issue by aardrian** [w3c/csswg-drafts#13784]
Reading through [§ 5. Accessibility Considerations](https://www.w3.org/TR/css-image-animation-1/#a11y) I see these statements:

> Currently, screen readers typically do not chose to announce animated images differently from non-animated images, though this could be implemented if found desirable. Similarly, it is not expected that they would announce paused or playing images any differently, though this could be implemented if found desirable.

Screen readers get their information from the user agent. The user agent does this by using platform accessibility APIs. How UAs convey semantics to platform AAPIs is covered by the [Core-AAM spec](https://www.w3.org/TR/core-aam-1.2/).

In order to ensure screen readers _can_ convey this information, the appropriate states (playing, paused) and property (animated) needs to come from the UA, which needs to be defined through this specification and likely a combination of Core-AAM and [ARIA](https://www.w3.org/TR/wai-aria-1.3/).

As of now, there is no [global property in ARIA](https://w3c.github.io/aria/#global_states) that maps to what this spec currently defines through the [`:animated-image`](https://www.w3.org/TR/css-image-animation-1/#selectordef-animated-image) pseudo-class.

There are also no [global states in ARIA](https://w3c.github.io/aria/#global_states) that map to the property values [`paused`](https://www.w3.org/TR/css-image-animation-1/#valdef-image-animation-paused) / [`stopped`](https://www.w3.org/TR/css-image-animation-1/#valdef-image-animation-stopped) nor [`running`](https://www.w3.org/TR/css-image-animation-1/#valdef-image-animation-running) / [`normal`](https://www.w3.org/TR/css-image-animation-1/#valdef-image-animation-normal).

If we look at [ARIA widget attributes](https://w3c.github.io/aria/#attrs_widgets), the closest one to convey playing or not might be [`aria-pressed`](https://w3c.github.io/aria/#aria-pressed), but that would require the image to be a button unless we redefine `aria-pressed`. It doesn't solve conveying that the image is animated. Trying to hack the exposure of the control type (animated image versus button) using [`aria-roledescription`](https://w3c.github.io/aria/#aria-roledescription) will likely cause confusion (and failure to localize), so that's not an option (IMO). A new widget role can possibly address that as well.

At the risk of speaking on behalf of the ARIA WG, the Core-AAM editors, browser makers, and platform makers, it might be worth scoping a new widget role and/or new ARIA properties so this can be conveyed to users.

That can also get UAs to develop native UIs to let users handle this (I know at least Firefox used to allow `Esc` to stop animated GIFs, so there may still be code in there to resurrect). That would also make voice control easier since they could define a common name (absent one in a spec, maybe via a widget role).

I see no reason to delay native UA controls when the accessibility support must be baked in to this feature from the start.

**Comment by airbr** [comment]
Just leaving a comment that I am engaged with this topic and learning a great deal about it, and support further work in this area and resulting decisions being documented and widely shared. I think its basically an issue of popular concern and the gravitas of the consideration here is notable.

**Comment by frivoal** [comment]
The question of native UI is IMO separate (and possibly addressed through https://github.com/whatwg/html/issues/12318), but exposing whether the image is animatable, and if so, what it's play state is, makes sense to me.

I suspect we don't need to distinguish between paused vs stopped, nor between running vs normal, as from a sighted user's point of view, either the animation is running, or it is not, so I am not sure the AAPIs need to convey more than that.

I am a little puzzled that the relevant states don't exist already though. Isn't that a problem that was already encountered with `<video>`?

**Comment by aardrian** [comment]
> I suspect we don't need to distinguish between paused vs stopped, nor between running vs normal, as from a sighted user's point of view, either the animation is running, or it is not, so I am not sure the AAPIs need to convey more than that.

Agreed, though not because of sighted users. Animated GIFs are functionally on or off. 

> I am a little puzzled that the relevant states don't exist already though. Isn't that a problem that was already encountered with `<video>`?

[Videos are essentially containers in Core-AAM](https://www.w3.org/TR/html-aam/#el-video), where the controls are exposed based on control type (so the _play_ button is a `button` role). When I last checked in 2023, [screen readers exposed those states well](https://adrianroselli.com/2023/09/browser-video-players-review.html#SRs).

While [CSS has `:playing` and `:paused` pseudo-classes](https://drafts.csswg.org/selectors/#selectordef-playing), those also have no mapping (ostensibly because of the video player controls — at this point I'm just speculating on a Saturday morning in an effort to answer your question).

**Comment by frivoal** [comment]
I have made some clarification to the statement in the specification: https://github.com/w3c/csswg-drafts/commit/d4fd9b01a87356432dcba44cf5bb9a7f0526960b

Beyond that, I am actually tempted to treat this request as orthogonal to the feature-set being described in the current specification.

* if the control attribute is specified on videos, the state of the various buttons can inform users about the state of the video, but as far as I understand, if the control attribute is not present, then videos do not indicate their playback state to the accessibility tree.
* web pages can already contain images that are animated, and images that are not. This is not exposed to the accessibility tree.

This isn't to say that exposing this information to the accessibility tree would not be useful, but that the existence of things that are playing and other things that are not playing predates the features being defined here, and seems possible to address largely independently of the `image-amimation` property. As you mentioned above @aardrian, [CSS has `:playing` and `:paused` pseudo-classes](https://drafts.csswg.org/selectors/#selectordef-playing), and I suspect we should probably convey the state of playing/paused video or images or other things from accessibility mappings tied to that rather than do something bespoke for images only.
